src/app/lib/business/environment/environment_manager.dart

```diff
@@ -0,0 +1,70 @@
+import 'package:app/access/environment/environment_repository.dart';
+import 'package:flutter_dotenv/flutter_dotenv.dart';
+
+/// Service to use as the source of truth for application environment (e.g. Staging, Production, etc.).
+abstract interface class EnvironmentManager {
+  factory EnvironmentManager(EnvironmentRepository repository, DotEnv dotEnv) =
+      _EnvironmentManager;
+
+  /// Gets the current environment (which can be overridden).
+  String get current;
+
+  /// Gets the next environment that will become the current one when the application restarts.
+  /// This value changes when using [setEnvironment].
+  String get next;
```

Comment:
Should this be nullable instead? If no change has been made for example? We could probably save a couple of bytes of memory by not assigning this string.